### DISCUSSION TITLE
To checksum or not to checksum?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/merkle-api#27] [General]
Questions for you @razgraf and @gavriliumircea based on the context below (from https://github.com/sablier-labs/v2-interfaces/issues/830):

1. Users are not forced to checksum the addresses, right? It only means that we will always check the addresses before generating the Merkle tree.
2. Will there be implications for the claim front-end, i.e., will all UIs have to provide the checksummed address?

<img width="600" alt="Screenshot 2024-01-18 at 6 26 17 PM" src="https://github.com/sablier-labs/v2-merkle-api/assets/8782666/294ecb59-bc76-4398-8a17-6b92eaa2437a">

**Comment by gavriliumircea** [comment]
1. No, users are not forced to checksum the addresses. Indeed before generating the Merkle tree we check if all the addresses are checksummed, and we checksum  the addresses the are not already.
2. For the eligibility call it does not matter if the address is checksummed or not(ONLY if the UI is using our backend and IPFS recommend structure)

For the now the process involving the checksumming didn't impact significantly the performance but this might change when we are going to work with very large campaigns.

**Reply by PaulRBerg** [reply]
very nice, thanks for explaining
